src/classes/class-pluginmanager.php

```diff
@@ -75,61 +67,41 @@ public function __construct() {
 		 * Initialise the magic.
 		 */
 		public function init() {
-			// Load class strings.
-			$this->strings = [
-				/* translators: 1: plugin name(s). */
-				'notice_can_install_required'  => _n_noop(
-					'WooCart requires the following plugin: %1$s.',
-					'WooCart requires the following plugins: %1$s.',
-					'woocart-defaults'
-				),
-				/* translators: 1: plugin name(s). */
-				'notice_ask_to_update'         => _n_noop(
-					'The following plugin needs to be updated to its latest version to ensure maximum compatibility with WooCart: %1$s.',
-					'The following plugins need to be updated to their latest version to ensure maximum compatibility with WooCart: %1$s.',
-					'woocart-defaults'
-				),
-				/* translators: 1: plugin name(s). */
-				'notice_ask_to_update_maybe'   => _n_noop(
-					'There is an update available for: %1$s.',
-					'There are updates available for the following plugins: %1$s.',
-					'woocart-defaults'
-				),
-				/* translators: 1: plugin name(s). */
-				'notice_can_activate_required' => _n_noop(
-					'The following required plugin is currently inactive: %1$s.',
-					'The following required plugins are currently inactive: %1$s.',
-					'woocart-defaults'
-				),
-			];
-
 			// Check for the admin panel.
 			if ( true !== ( is_admin() && ! defined( 'DOING_AJAX' ) ) ) {
 				return;
 			}
 
 			// Register plugins.
 			foreach ( $this->list as $plugin ) {
-				// @codeCoverageIgnoreStart
-				$this->register( $plugin );
-				// @codeCoverageIgnoreEnd
+				/**
+				 * Special case for Sendgrid.
+				 * We register the plugin only if the the `SENDGRID_API_KEY` is defined.
+				 *
+				 * @see https://github.com/niteoweb/woocart-docker-web/blob/master/fixtures/config/wp-config.php
+				 */
+				if ( 'sendgrid-email-delivery-simplified' === $plugin['slug'] ) {
```

Comment:
Remove this, we need to handle this in the consumer of this feature, this package must not have any operational knowledge.